javaslang/src/main/java/javaslang/collection/BitMappedTrie.java

```diff
@@ -177,11 +177,18 @@ T get(int index) {
      */
     @SuppressWarnings("unchecked")
     T[] getLeaf(int index) {
-        Object[] leaf = array;
-        for (int shift = depthShift; shift > 0; shift -= BRANCHING_BASE) {
-            leaf = getAt(leaf, digit(offset + index, shift));
+        if (depthShift == 0) {
+            return (T[]) array;
+        } else if (depthShift == BRANCHING_BASE) {
+            return getAt(array, firstDigit(offset + index, depthShift));
+        } else {
```

Comment:
The else branch covers `depthShift > 0 && (depthShift < BRANCHING_BASE || depthShift > BRANCHING_BASE)`, right?

Just asking to get sure that `depthShift` can be less than `BRANCHING_BASE`...